bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTableViewer.java

```diff
@@ -269,6 +269,13 @@ public void add(Object[] elements) {
 			return;
 		Object[] filtered = filter(elements);
 
+		// if limit is set and still some elements are not populated. Assumption that
+		// user has already updated the model and needs addition of an item.
+		if (getItemsLimit() > 0 && filtered.length > 0 && getLastElement() instanceof IExpandableNode) {
+			internalRefreshAll(false);
```

Comment:
This does not scale :-(
Try Search view (set to "table" and without limit) and plain text search with `a` in a big workspace. 
`FileTableContentProvider.elementsChanged(Object[])` adds elements one by one, and we constantly fetch/sort/label ~25.000 elements...

I wonder if we can just add the element to the "expanded node" array and sort it, to avoid constant refreshes for each element.